### PR TITLE
Update atuin to v18.1.0

### PR DESCRIPTION
**Pull request by papertigers**
Version 17.0.0 just released: https://github.com/atuinsh/atuin/releases/tag/v17.0.0

I haven't run through a build of this, but I am filing the PR anyways.

### CODE REVIEW

**Review by oetiker — DISMISSED** [review]

File: build/atuin/build.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@
 . ../../lib/build.sh
 
 PROG=atuin
-VER=16.0.0
+VER=17.0.0
```
Comment (hadfl) [inline build/atuin/build.sh:20]: In the meantime version `17.0.1` has been released. Would you mind bumping it to the latest release?

**Review by hadfl — CHANGES_REQUESTED** [review]

**Review by hadfl — CHANGES_REQUESTED** [review]
Also the build seems to fail, we'd appreciate if you could have a look and fix it if possible. Thanks

**Comment by oetiker** [comment]
any progress ?

**Comment by papertigers** [comment]
Sorry haven't had time to put together the necessary patch on the atuin side.  
Let me see if I can make some time soon to push that along.

**Comment by citrus-it** [comment]
18.1.0 fixes this, and also fixes some undefined behaviour on illumos. I've updated the PR.

**Review by hadfl — APPROVED** [review]

**Review by citrus-it — APPROVED** [review]

**Pull request merged**